### ISSUE TITLE
False positive - `no-handler: Tasks that run when changed should likely be handlers.`

### ISSUE THREAD

**Issue by ReenigneArcher** [ansible/ansible-lint#3655]
##### Summary

Ansible lint complains of `no-handler: Tasks that run when changed should likely be handlers.` when it is already a handler.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

Using ansible/ansible-lint-action@v6.17.0

Unable to use new action at this time due to #3648 

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package
GitHub action.

##### STEPS TO REPRODUCE

```yml
- name: Test
  hosts:
    - test
  gather_facts: false
  connection: local

  tasks:
    - name: Get info
      delegate_to: localhost
      register: collected_info
      somemodule.that.collects.info:

    - name: Do something
      delegate_to: localhost
      loop: "{{ collected_info['some_list'] }}"
      loop_control:
        label: "{{ item.name }}"
      notify:
        - Debug
      register: _something_done
      somemodule.that.may.change.something:

  handlers:
    - name: Debug
      loop: "{{ _something_done.results }}"
      loop_control:
        label: "{{ item.item.name }}"
      when: item.changed
      ansible.builtin.msg:
        msg: "{{ item.item.name }} changed"

```

##### Desired Behavior

Don't warn that something should be a handler that already is a handler.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ajakk** [comment]
I'm hitting this too: https://github.com/ajakk/ansible/actions/runs/5919432937

The check isn't really wrong, but a more intelligent check might be taught that `var.changed` is a fine thing to check in handlers when `loop`ing, or that handlers shouldn't be treated as tasks for the purposes of this check, or both.

**Comment by ajakk** [comment]
This fix is incomplete in my case, where the handlers are actually in a [separate tasks file](https://github.com/ajakk/ansible/blob/main/roles/automount/handlers/main.yml). The new `Task.is_handler` seems to return false in this case.

Adding a `pdb` invocation in no_handler.py:
```
diff --git a/src/ansiblelint/rules/no_handler.py b/src/ansiblelint/rules/no_handler.py
index 165bbf43..8ce4a4a9 100644
--- a/src/ansiblelint/rules/no_handler.py
+++ b/src/ansiblelint/rules/no_handler.py
@@ -69,6 +69,7 @@ class UseHandlerRatherThanWhenChangedRule(AnsibleLintRule):
         task: Task,
         file: Lintable | None = None,
     ) -> bool | str:
+        import pdb; pdb.set_trace()
         if task["__ansible_action_type__"] != "task" or task.is_handler():
             return False
```

```
(Pdb) task
Task('Unmount filesystems' [.[0]])
(Pdb) task.position
'.[0]'
(Pdb) task.is_handler()
False
```
